textbook/chapter_1.qmd

```diff
@@ -0,0 +1,993 @@
+
+## ggplot
+
+```{python}
+#| echo: false
+#| output: false
+%%R
+suppressPackageStartupMessages(library(tidyverse))
+```
+
+```{python}
+%%R
+library(tidyverse)
+theme_set(theme_gray(base_size = 14))
+
+url = 'https://raw.githubusercontent.com/joelostblom/teaching-datasets/main/cars.csv'
+cars = read_csv(url, show_col_types = FALSE)
+cars
+```
+:::
+
+
+::: {.column-margin}
+
+<strong> Stellar Charts </strong>
+<br>
+Altair is the name of a star in the same constellation as the star Vega, which is also the name of the Javascript library that Altair is built upon. Meanwhile, the "gg" in ggplot is a reference to the book "Grammar of Graphics",
+which inspired the creation of the ggplot library. 
+
+<br>
+![](https://upload.wikimedia.org/wikipedia/commons/6/69/Altair.jpg)
+
+:::
+
+The fundamental object in Altair is the `Chart`, which takes a data frame as a single argument `alt.Chart(cars)`.
+
+With a chart object in hand, we can now specify how we would like the data to be visualized. We first indicate what kind of geometric *mark* we want to use to represent the data. We can set the `mark` attribute of the chart object using the the `Chart.mark_*` methods.
+
+For example, we can show the data as a point using `mark_point()`:
+
+
+```{python}
+alt.Chart(cars).mark_point()
+```
+
+
+Here the rendering consists of one point per row in the dataset, all plotted on top of each other, since we have not yet specified positions for these points.
+
+To visually separate the points, we can map various *encoding channels*, or *channels* for short, to fields in the dataset. For example, we could *encode* the field `Miles_per_Gallon` of the data using the `x` channel, which represents the x-axis position of the points. To specify this, use the `encode` method:
+
+
+::: {.panel-tabset}
+
+## Altair
+```{python}
+alt.Chart(cars).mark_point().encode(
+    x='Miles_per_Gallon'
+)
+```
+
+## ggplot
+
+
+```{python}
+%%R
+ggplot(cars, aes(x = Miles_per_Gallon, y=0)) +
+  geom_point()
+```
+
+::: {.column-margin}
+
+<br>
+<br>
+
+In ggplot2, you begin by creating a plot object with your dataset, such as `ggplot(cars)`. This is similar to Altair's `Chart(cars)`. To visualize the data, you add a geometric object like `geom_point()`, which is conceptually equivalent to Altair's `mark_point()`. Initially, this would plot all points at the origin. To meaningfully represent the data, you map variables to visual properties using `aes()`. For instance, to plot Miles per Gallon on the x-axis, you'd use `aes(x = Miles_per_Gallon)` within `geom_point()`. This corresponds to Altair's `encode` method. 
+:::
+
+:::
+The `encode()` method builds a key-value mapping between encoding channels (such as `x`, `y`, `color`, `shape`, `size`, *etc.*) to fields in the dataset, accessed by field name. For Pandas data frames, Altair automatically determines an appropriate data type for the mapped column, which in this case is the *nominal* type, indicating unordered, categorical values.
```

Comment:
Interesting solution with putting the text in the margin here. But the paragraph under the chart keeps talking about the altair syntax. Maybe that is ok and it is too difficult to switch all the text according the the programming langauge selected

Reply:
Yes this was a bit of a judgement call. I found that it wasn't easy to say 'this is the Altair paragraph associated with the visual' as the paragraphs flow from previous ones. (and the subsequent ones often depends on those ones too). It seemed to make more sense to add a ggplot paragraph that replace an existing one as it wasn't a one-to-one match. I tried adding above and below within the tabbed area, but above changes the chart position and below is difficult to notice (as it's often below the 'fold' of the page so you need to scroll down).

Reply:
But I'm open to whatever solution you come up with. One thing that could help is making the sidebar text a bit more prominent if we do opt for that approach.

Reply:
I have attempted a solution where we use tabs for all package related content and margins for optional tidbits/clarifications (details in the commit messages).